shims/common/src/main/java/org/apache/hadoop/hive/thrift/HadoopThriftAuthBridge.java

```diff
@@ -372,6 +389,46 @@ public TTransportFactory createTransportFactory(Map<String, String> saslProps)
     }
 
     /**
+     * Create a TTransportFactory that, upon connection of a client socket,
+     * negotiates a Kerberized SASL transport with PLAIN mechanism.
+     * The resulting TTransportFactory can be passed as both the input and output
+     * transport factory when instantiating a TThreadPoolServer, for example.
+     *
+     * @param saslProps Map of SASL properties
+     */
+    public TTransportFactory createPlainTransportFactory(Map<String, String> saslProps)
+        throws TTransportException {
+
+      TSaslServerTransport.Factory transFactory = createSaslPlainServerTransportFactory(saslProps);
+
+      return new TUGIAssumingTransportFactory(transFactory, realUgi);
+    }
+
+    /**
+     * Create a TSaslServerTransport.Factory that, upon connection of a client
+     * socket, negotiates a Kerberized SASL transport.
```

Comment:
would it be worthwhile to mention the fact that this method requires the customAuthClass to be configured?